tests/components/insteon/test_config_flow.py

```diff
@@ -0,0 +1,703 @@
+        assert entry.data[CONF_HOST] == MOCK_HOSTNAME
+        assert entry.data[CONF_PORT] == MOCK_PORT
+        assert entry.data[CONF_USERNAME] == MOCK_USERNAME
+        assert entry.data[CONF_PASSWORD] == MOCK_PASSWORD
+        assert entry.data[CONF_HUB_VERSION] == 2
+        assert CONF_OVERRIDE not in entry.data
+        assert CONF_X10 not in entry.data
+        assert entry.options[CONF_OVERRIDE][0][CONF_ADDRESS] == "1A.2B.3C"
+        assert entry.options[CONF_X10][0][CONF_HOUSECODE] == MOCK_HOUSECODE
+
+
+async def test_import_min_hub_v2(hass: HomeAssistantType):
+    """Test importing a minimum Hub v2 config from yaml."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    result = await _import_config(hass, MOCK_IMPORT_MINIMUM_HUB_V2)
+
+    assert result["type"] == "create_entry"
+    assert hass.config_entries.async_entries(DOMAIN)
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        assert entry.data[CONF_HOST] == MOCK_HOSTNAME
+        assert entry.data[CONF_PORT] == 25105
+        assert entry.data[CONF_USERNAME] == MOCK_USERNAME
+        assert entry.data[CONF_PASSWORD] == MOCK_PASSWORD
+        assert entry.data[CONF_HUB_VERSION] == 2
+
+
+async def test_import_min_hub_v1(hass: HomeAssistantType):
+    """Test importing a minimum Hub v1 config from yaml."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    result = await _import_config(hass, MOCK_IMPORT_MINIMUM_HUB_V1)
+
+    assert result["type"] == "create_entry"
+    assert hass.config_entries.async_entries(DOMAIN)
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        assert entry.data[CONF_HOST] == MOCK_HOSTNAME
+        assert entry.data[CONF_PORT] == 9761
+        assert entry.data[CONF_HUB_VERSION] == 1
+
+
+async def test_import_existing(hass: HomeAssistantType):
+    """Test we fail on an existing config imported."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        entry_id="abcde12345",
+        data={**MOCK_USER_INPUT_HUB_V2, CONF_HUB_VERSION: 2},
+        options={},
+    )
+    config_entry.add_to_hass(hass)
+    assert config_entry.state == config_entries.ENTRY_STATE_NOT_LOADED
+
+    result = await _import_config(hass, MOCK_IMPORT_MINIMUM_HUB_V2)
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "already_configured"
+
+
+async def test_import_failed_connection(hass: HomeAssistantType):
+    """Test a failed connection on import."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    with patch(PATCH_CONNECTION, new=mock_failed_connection,), patch(
+        PATCH_ASYNC_SETUP, return_value=True
+    ), patch(PATCH_ASYNC_SETUP_ENTRY, return_value=True):
+        result = await async_import_config(hass, MOCK_IMPORT_MINIMUM_HUB_V2)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "cannot_connect"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)